reconcile/aus/base.py

```diff
@@ -615,112 +583,118 @@ def version_conditions_met(
 
 
 def gates_to_agree(
-    version_prefix: str, cluster: str, cluster_version: str, ocm: OCM
+    gates: list[OCMVersionGate],
+    version_prefix: str,
+    cluster: OCMCluster,
+    ocm_api: OCMBaseClient,
 ) -> list[str]:
     """Check via OCM if a version is agreed
 
     Args:
+        gates: list of OCMVersionGate objects
         version_prefix (string): major.minor version prefix
         cluster (string)
         cluster_version (string): current version of the cluster
-        ocm (OCM): used to fetch infos from OCM
+        sts (bool): is the cluster a STS cluster
+        ocm_api (OCMBaseClient): used to fetch infos from OCM
 
     Returns:
         list[str]: list of gate ids to agree
     """
-    agreements = {
-        agreement["version_gate"]["id"]
-        for agreement in ocm.get_version_agreement(cluster)
-    }
-    semver_cluster = parse_semver(f"{cluster_version}")
-
-    return [
-        gate["id"]
-        for gate in ocm.get_version_gates(version_prefix)
-        if gate["id"] not in agreements and semver_cluster.match(f"<{version_prefix}.0")
+    semver_cluster = parse_semver(f"{cluster.version.raw_id}")
+
+    relevant_gates = [
+        g
+        for g in gates
+        if g.version_raw_id_prefix == version_prefix
+        and g.sts_only == cluster.is_sts()
```

Comment:
❤️

Reply:
hmm i am actually not sure if we should only retrieve `sts_only` gates for STS clusters, or if all gates should be checked int that case

Reply:
that's only a guess. I think `only_sts` refers to the gate, meaning that gate applies only to STS clusters. But other gates may apply to all, including STS. We should confirm with OCM folks I think.

Reply:
your guess is right.
given the bigger risk of version gates for STS clusters, i opened a dedicated ticket to discuss a potential change for the version gate agreement flow - https://issues.redhat.com/browse/APPSRE-7949

Reply:
with [9c865a5](https://github.com/app-sre/qontract-reconcile/pull/3655/commits/9c865a5fee29d5c782b95ab6a8447b4d93223d22) sts-only gates are ignored and not approved. this is compatible with how it worked before the refactor.

but if we would start to do y-stream upgrades that usually involve an sts-version gate, scheduling upgrades for sts-clusters will fail. this is where https://issues.redhat.com/browse/APPSRE-7949 needs to be implemented to handle it safely